### ISSUE TITLE
[@angular-eslint/template/no-call-expression] support no assignment expression

### ISSUE THREAD

**Issue by deerawan** [angular-eslint/angular-eslint#940]
**Description and reproduction of the issue**

To enforce immutability on the template especially for `any`, define a way to prevent assignment expression. 

```JSON
{
  "rules": {
    "@angular-eslint/template/no-call-expressioin": ["error", { "no-assignment": true }]
  }
}
```

For component with `OnPush` strategy, immutability is quite important. In Typescript, we could encourage this paradigm by using `Readonly` for object. 

```ts
interface Person { 
  readonly name: string;
  readonly age: number;
}

john: Person = {
    name: 'john',
    age: 17
}
```

It works very well on the template to detect mutation as below

HTML
```html
<div (click)="john.name = 'john doe'"></div> <!-- Cannot assign to 'name' because it is a read-only property. -->
```

The problem is if you have `any` type for a specific field that allows object

```ts
interface Person { 
  readonly name: string;
  readonly age: number;
  readonly data: any;
}

john: Person = {
  name: 'john',
  age: 17,
  address: {
    city: 'SF'
  }
}
```

HTML
```html
<div (click)="john.address.city = 'LA'"></div> <!-- mutation is not detected -->
```

Maybe this should come as new rule e.g `no-assignment-expression` 

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `12.3.1` |
| `@angular-eslint/template-parser`         | `12.3.1` |
| `@typescript-eslint/parser`               | `4.28.2` |
| `ESLint`                                  | `7.26.0` |
| `node`                                    | `14.17.0` |

```sh
# Please run `npx ng version` in your project and paste the full output here:

```
Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1200.5
@angular-devkit/build-angular   12.0.5
@angular-devkit/core            12.0.5
@angular-devkit/schematics      12.0.5
@schematics/angular             12.0.5
rxjs                            6.6.7
typescript                      4.2.4
```
